### ISSUE TITLE
[BUG] Jackson afterburner warning messages are logged when first exception occurs

### ISSUE THREAD

**Issue by basclaessen** [Azure/azure-sdk-for-java#19847]
**Describe the bug**
Jackson afterburner warning messages when first exception occurs. Related to #18215

**To Reproduce**
Create a cosmos client and force an exception to occur.

***Exception or Stack Trace***
```
Mar 15, 2021 8:37:23 AM com.fasterxml.jackson.module.afterburner.ser.OptimizedBeanPropertyWriter _reportProblem
WARNING: Disabling Afterburner serialization for class com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics (field #0; mutator com.fasterxml.jackson.module.afterburner.ser.StringMethodPropertyWriter), due to access error (type java.lang.IllegalAccessError, message=failed to access class com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics from class com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics$Access4JacksonSerializer254c895f (com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics is in unnamed module of loader 'app'; com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics$Access4JacksonSerializer254c895f is in unnamed module of loader com.fasterxml.jackson.module.afterburner.util.MyClassLoader @1b0a7baf))

java.lang.IllegalAccessError: failed to access class com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics from class com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics$Access4JacksonSerializer254c895f (com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics is in unnamed module of loader 'app'; com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics$Access4JacksonSerializer254c895f is in unnamed module of loader com.fasterxml.jackson.module.afterburner.util.MyClassLoader @1b0a7baf)
	at com.azure.cosmos.implementation.ClientSideRequestStatistics$GatewayStatistics$Access4JacksonSerializer254c895f.stringGetter(com/azure/cosmos/implementation/ClientSideRequestStatistics$GatewayStatistics$Access4JacksonSerializer.java)
	at com.fasterxml.jackson.module.afterburner.ser.StringMethodPropertyWriter.serializeAsField(StringMethodPropertyWriter.java:44)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:3126)
	at com.fasterxml.jackson.core.base.GeneratorBase.writeObject(GeneratorBase.java:388)
	at com.fasterxml.jackson.core.JsonGenerator.writeObjectField(JsonGenerator.java:2332)
	at com.azure.cosmos.implementation.ClientSideRequestStatistics$ClientSideRequestStatisticsSerializer.serialize(ClientSideRequestStatistics.java:303)
	at com.azure.cosmos.implementation.ClientSideRequestStatistics$ClientSideRequestStatisticsSerializer.serialize(ClientSideRequestStatistics.java:264)
	at com.fasterxml.jackson.databind.SerializerProvider.defaultSerializeValue(SerializerProvider.java:1118)
	at com.fasterxml.jackson.databind.node.POJONode.serialize(POJONode.java:115)
	at com.fasterxml.jackson.databind.node.ObjectNode.serialize(ObjectNode.java:328)
	at com.fasterxml.jackson.databind.ser.std.SerializableSerializer.serialize(SerializableSerializer.java:39)
	at com.fasterxml.jackson.databind.ser.std.SerializableSerializer.serialize(SerializableSerializer.java:20)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4487)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:3742)
	at com.azure.cosmos.CosmosException.toString(CosmosException.java:342)
	at java.base/java.lang.String.valueOf(String.java:2951)
	at java.base/java.io.PrintStream.println(PrintStream.java:897)
	at java.base/java.lang.Throwable$WrappedPrintStream.println(Throwable.java:752)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:659)
	at java.base/java.lang.Throwable.printStackTrace(Throwable.java:648)
	at java.base/java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1059)
	at java.base/java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1050)
	at java.base/java.lang.Thread.dispatchUncaughtException(Thread.java:2002)
```

***Code Snippet***
```java
import com.azure.cosmos.CosmosAsyncClient;
import com.azure.cosmos.CosmosAsyncContainer;
import com.azure.cosmos.CosmosAsyncDatabase;
import com.azure.cosmos.CosmosClientBuilder;
import com.azure.cosmos.models.PartitionKey;
import com.fasterxml.jackson.databind.JsonNode;

public class CosmosAfterburnerWarning {
    public static void main(String[] args) throws Exception {
        String endpoint = "YOUR_COSMOS_ENDPOINT";
        String key = "YOUR_COSMOS_KEY";

        CosmosAsyncClient cosmosAsyncClient = new CosmosClientBuilder()
                .endpoint(endpoint)
                .key(key)
                .buildAsyncClient();

        //force an exception
        CosmosAsyncDatabase database = cosmosAsyncClient.getDatabase("non_existing_database");
        CosmosAsyncContainer container = database.getContainer("non_existing_container");
        container.readItem("itemId", new PartitionKey("partitionKey"), JsonNode.class).block();

    }
}
```
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>test.cosmos</groupId>
    <artifactId>test.cosmos</artifactId>
    <version>0.0.1-SNAPSHOT</version>

    <properties>
        <maven.compiler.target>11</maven.compiler.target>
        <maven.compiler.source>11</maven.compiler.source>
    </properties>

    <dependencies>
        <dependency>
            <groupId>com.azure</groupId>
            <artifactId>azure-cosmos</artifactId>
            <version>4.13.0</version>
        </dependency>
    </dependencies>
</project>
```

**Expected behavior**
This warning should not be logged

**Setup (please complete the following information):**
 - OS: Linux & Windows
 - IDE : IntelliJ
 - azure-cosmos library version : 4.13.0

**Additional context**
No additional context

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Bug Description Added
- [X] Repro Steps Added
- [X] Setup information Added

**Comment by basclaessen** [comment]
@kushagraThapar The issue still occurs with azure-cosmos version 4.13.1. Can you reopen this issue please?

**Comment by kushagraThapar** [comment]
@basclaessen - Yes, surprisingly I am able to reproduce this issue again. I will re-open this issue and will fix it soon. Thanks for raising it again.

**Comment by kushagraThapar** [comment]
@basclaessen - The reason I was able to fix it is because I tried your repro and it was not showing the warning after my fix, using locally built jar from my local .m2 cache.  

But when I try the released version (4.13.1) - then it the issue appears again. I will investigate more on this.
